### DISCUSSION TITLE
Public/private access

### DISCUSSION THREAD

**Discussion by quintennoels** [Azure/ipam#189] [Ideas]
Hi, 
Could we add a section to the documentation on how to make the app as privately accessible as possible?
- Private endpoint for inbound access to the webapp + turn public access off
- Keyvault/cosmosdb only accessible through selected IPs (the outbound public ips of the webapp)
- ... Am I forgetting anything?
Thanks!

**Comment by DCMattyG** [comment]
Hi @quintennoels, we have been working on some docs for this as it is a top ask. That said, in the deployments current form it uses a multi-container App Service which doesn't support vNET Integration (at this point in time).

I'm currently in the process (as we speak) of re-working the app to work as either a single container or native code running in an App Service and this will enable the paradigms that you've lined out above.

I'll definitely include the "how to make everything private" docs shortly after the next upcoming release and that should help get everyone where they need to be 👍

**Reply by quintennoels** [reply]
Hi Matty,
Thanks for your quick reply.
Great to hear it's coming.

Any idea on the timeline? :)

**Reply by DCMattyG** [reply]
I'm targeting the end of the month for these updates @quintennoels.

**Reply by Dylan-Prins** [reply]
Hi @DCMattyG,

How is this topic going?

**Reply by peyton-smith-kh** [reply]
Hey @DCMattyG I've been following this discussion is there any update on this by chance? Thanks!

**Reply by DCMattyG** [reply]
Hi @peyton-smith-kh and @Dylan-Prins, it's taking me a bit longer to finish the updates to the code, test everything, and make sure the documentation is reflective of the fairly major changes that will be involved. I'll have 2 solid weeks this month where I can focus on getting all of those items complete, so I can commit to having all of these items completed by the end of the calendar year (~30 days). You'll see the update code & documentation by 1/1/2024.

I do apologize for the delay but collapsing all of this code is taking a bit more time than expected, mostly because I want to make sure everything is perfect for the new release and there is an upgrade process for those users on a previous deployment type so folks can seamless migrate to the updated platform.

**Reply by peyton-smith-kh** [reply]
@DCMattyG I sincerely appreciate the quick response! Thank you for your efforts!

**Reply by omerzubair** [reply]
Thanks heaps for the update @DCMattyG. Our organisation will not let us deploy without privateendpoint and appgateway Infront. While GUI is awesome but for our usage a simple function app with api calling with privateendpoint + appgateway in front will be just fine. 
Looking forward Jan 2024 :)

**Reply by omerzubair** [reply]
Happy new year @DCMattyG , Hope you had a wonderful break. 
looking forward to the new code drop.
alternatively, any diagram will be super useful as well.

**Reply by DCMattyG** [reply]
You bet @omerzubair. Please keep in mind that the architecture really doesn't change much. The primary advantages will be that end users can anchor these services privately into their vNET now much more easily. I will include some docs on possible ways to accomplish this, but at the end of the day there are numerous ways in which this can be accomplished and will vary slightly per customer environment.

**Reply by vladimirs-davidovs** [reply]
Hello @DCMattyG!
I wanted to ask if there are any news related to the code/documentation update, discussed in this thread? We are hardening our environment at the moment and enabling vNet integration for the IPAM Web Apps would be really important.

**Reply by Dylan-Prins** [reply]
Hi @DCMattyG,

Is there any progress?

**Reply by Dylan-Prins** [reply]
Still nothing?!

**Reply by JBirSP** [reply]
> Still nothing?!

There doesn't seem to be anything in the docs, but I was successful in deploying with private endpoints and public access disabled using the steps described here: https://github.com/Azure/ipam/issues/244#issuecomment-1984250957

**Reply by DCMattyG** [reply]
@JBirSP thank you so much for linking that here and @Dylan-Prins my apologies for the continued delay on this item. There's been quite a lot going on with new features and some bugs fixes and I've unfortunately fallen behind on updating the docs. I did share a high-level guide in the above mentioned link and if you need some more personalized guidance until I'm able to complete the doc updates, please feel free to reach out to me via email at Matthew.Garrett@microsoft.com and I can provide you some more detailed screenshots and steps for now, which will eventually make it into a nice-formatted docs page.

Sorry again, just trying to make sure I'm properly prioritizing all of the amazing requests that have come in from the community 😄

**Reply by Dylan-Prins** [reply]
No problem! Can imagine you are busy.

Thanks for the update!
